src/components/input/input.js

```diff
@@ -634,11 +634,25 @@ function mdMaxlengthDirective($animate, $mdUtil) {
   };
 
   function postLink(scope, element, attr, ctrls) {
-    var maxlength;
+    var maxlength = Number.parseInt(attr.mdMaxlength);
+    if (isNaN(maxlength)) maxlength = -1;
```

Comment:
Should this use `!angular.isNumber(maxlength)` to be consistent with line 645, or is there a reason that `isNaN` better here?

Reply:
No, that's not equivalent here, because angular.isNumber(NaN) returns true but in this case, we do not want to keep NaN.

Reply:
Ah yes, I thought that `angular.isNumber()` was smarter than it actually is.